src/assessments/seo/largestKeywordDistanceAssessment.js

```diff
@@ -0,0 +1,133 @@
+const AssessmentResult = require( "../../values/AssessmentResult.js" );
+const Assessment = require( "../../assessment.js" );
+const merge = require( "lodash/merge" );
+const countWords = require( "../../stringProcessing/countWords.js" );
+const matchWords = require( "../../stringProcessing/matchTextWithWord.js" );
+const Mark = require( "../../values/Mark.js" );
+const marker = require( "../../markers/addMark.js" );
+const inRangeStartEndInclusive = require( "../../helpers/inRange.js" ).inRangeStartEndInclusive;
+
+/**
+ * Returns a score based on the largest percentage of text in
+ * which no keyword occurs.
+ */
+class largestKeywordDistanceAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		const defaultConfig = {
+			overRecommendedMaximumKeywordDistance: 40,
+			recommendedMaximumKeywordDistance: 30,
+			scores: {
+				good: 9,
+				okay: 6,
+				bad: 1,
+			},
+		};
+
+		this.identifier = "largestKeywordDistance";
+		this._config = merge( defaultConfig, config );
+	}
+
+	/**
+	 * Runs the largestKeywordDistance research and based on this returns an assessment result.
+	 *
+	 * @param {Paper}       paper       The paper to use for the assessment.
+	 * @param {Researcher}  researcher  The researcher used for calling research.
+	 * @param {Object}      i18n        The object used for translations.
+	 *
+	 * @returns {AssessmentResult} The assessment result.
+	 */
+	getResult( paper, researcher, i18n ) {
+		this._largestKeywordDistance = researcher.getResearch( "largestKeywordDistance" );
+		let assessmentResult = new AssessmentResult();
+
+		const calculatedResult = this.calculateResult( i18n );
+
+		assessmentResult.setScore( calculatedResult.score );
+		assessmentResult.setText( calculatedResult.resultText );
+		assessmentResult.setHasMarks( calculatedResult.score < 2 );
```

Comment:
Why not to allow markings for the OK case? I am afraid it is an artefact of the previously used scoring schema that only had good and bad.
Please also fix it in feature/recalibration.

Reply:
Fixed for this PR

Concerning `feature/recalibration`: I think once we'll have merged `feature/synonyms` into `feature/development` and `feature/development` into `feature/recalibration` that will take care of it.